dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityService.java

```diff
@@ -271,34 +268,11 @@ private List<TrackedEntity> findTrackedEntities(
     }
     trackedEntityAuditService.addTrackedEntityAudit(SEARCH, user.getUsername(), trackedEntities);
 
-    // TODO(tracker): Push this filter into the store because it is breaking pagination
-    return trackedEntities.stream()
-        .filter(filterAccessibleTrackedEntities(user, queryParams))
-        .toList();
+    return trackedEntities;
   }
 
   @Override
   public Set<String> getOrderableFields() {
     return trackedEntityStore.getOrderableFields();
   }
-
-  private Predicate<TrackedEntity> filterAccessibleTrackedEntities(
-      UserDetails user, TrackedEntityQueryParams queryParams) {
-    boolean skipOwnershipCheck = queryParams.getOrgUnitMode() == ALL;
-
-    if (queryParams.hasEnrolledInTrackerProgram()) {
-      return te ->
-          skipOwnershipCheck
-              || ownershipAccessManager.hasAccess(
-                  user, te, queryParams.getEnrolledInTrackerProgram());
-    }
-
-    return te ->
-        queryParams.getAccessibleTrackerPrograms().stream()
-            .filter(
```

Comment:
Do we have an existing test to reaches this filtering based on "any" program ownership?